service/s3/s3manager/bucket_region.go

```diff
@@ -86,3 +147,10 @@ func GetBucketRegionWithClient(ctx aws.Context, svc s3iface.S3API, bucket string
 
 	return bucketRegion, nil
 }
+
+func validateEndpointWithoutRegion(r *request.Request) {
+	if len(aws.StringValue(r.Config.Endpoint)) == 0 {
```

Comment:
Should this also look for len(r.ClientInfo.Endpoint)?

Reply:
The idea is to ensure that the behavior is only used if the caller explicitly provided an endpoint URL. vs `ClientInfo` in the other validator could also be the resolved endpoint URL. I'll add more docs inline to make these more clear.